shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Value.hs

```diff
@@ -157,20 +147,34 @@ instance Era era => Val (Value era) where
       addrHashLen = 28
 
 -- ============================================================================
--- Operations on Map, specialised to comparable `Monoid` values.
+-- Operations on Map, specialised to comparable `Monoidal` values.
+-- We need this because there is no non-Orphan (Monoid Integer) instance
```

Comment:
In the old code, when actually had a Val instance of Integer derived through Coin. I think this is much clearer. It says exactly what is required. I would use the Monoid class except
there is no (Monoid Integer) instance.

Reply:
`Sum` is here: http://hackage.haskell.org/package/base-4.14.0.0/docs/Data-Monoid.html#v:Sum

Reply:
The most exacting reason is in the instance for Map. Note we need maps that do not store the zero. Note:
instance (Eq k, Eq v, Ord k, Monoidal v) => Monoidal (Map k v) where
  zeroM = Map.empty
  plusM = unionWithV plusM

We cannot use the Monoid instance of Map, because the plusM is specific to maps that do not store the zero. So we need a new class regardless.
```